### PR TITLE
Implement new runtime_type for Turbopack

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Implementation of https://github.com/vercel/turbo/pull/7711

Closes NEXT-2791

### CODE REVIEW

**Comment by timneutkens** [comment]
* **#63212** <a href="https://app.graphite.dev/github/pr/vercel/next.js/63212?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/next-swc/crates/next-api/src/project.rs [hunk 1]
```diff
@@ -504,15 +505,16 @@ impl Project {
                 node_root.join("chunks".to_string()),
                 node_root.join("assets".to_string()),
                 node_build_environment(),
+                next_mode.runtime_type(),
```
Comment (sokra) [inline packages/next-swc/crates/next-api/src/project.rs:508]: The execution context is for postcss/webpack execution and I guess it should always use the dev runtime

File: packages/next-swc/crates/next-core/src/next_client/context.rs [hunk 2]
```diff
@@ -162,7 +161,6 @@ pub async fn get_client_resolve_options_context(
             Vc::upcast(ModuleFeatureReportResolvePlugin::new(project_path)),
             Vc::upcast(UnsupportedModulesResolvePlugin::new(project_path)),
             Vc::upcast(NextSharedRuntimeResolvePlugin::new(project_path)),
-            Vc::upcast(get_invalid_server_only_resolve_plugin(project_path)),
```
Comment (sokra) [inline packages/next-swc/crates/next-core/src/next_client/context.rs:165]: That looks unrelated to this change?

Comment (timneutkens) [reply packages/next-swc/crates/next-core/src/next_client/context.rs]: Yeah you're right... seems like a bad rebase

File: packages/next-swc/crates/next-api/src/project.rs (hunk 1 above)
Comment (timneutkens) [reply packages/next-swc/crates/next-api/src/project.rs:508]: Let's revisit that in the follow-up PR that introduces the production runtime as for now they both use the dev runtime anyway.

Curious to learn more about why this should use the dev runtime though

Comment (timneutkens) [reply packages/next-swc/crates/next-api/src/project.rs:508]: Opened a PR for this: https://github.com/vercel/turbo/pull/7721/files
